akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala

```diff
@@ -22,38 +24,46 @@ class FramingSpec extends AkkaSpec {
   val settings = ActorMaterializerSettings(system)
   implicit val materializer = ActorMaterializer(settings)
 
-  class Rechunker extends PushPullStage[ByteString, ByteString] {
-    private var rechunkBuffer = ByteString.empty
+  class Rechunker extends GraphStage[FlowShape[ByteString, ByteString]] {
 
-    override def onPush(chunk: ByteString, ctx: Context[ByteString]): SyncDirective = {
-      rechunkBuffer ++= chunk
-      rechunk(ctx)
-    }
+    val out: Outlet[ByteString] = Outlet("Rechunker.out")
+    val in: Inlet[ByteString] = Inlet("Rechunker.in")
+    override val shape: FlowShape[ByteString, ByteString] = FlowShape(in, out)
 
-    override def onPull(ctx: Context[ByteString]): SyncDirective = {
-      rechunk(ctx)
-    }
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+      new GraphStageLogic(shape) with InHandler with OutHandler {
 
-    override def onUpstreamFinish(ctx: Context[ByteString]): TerminationDirective = {
-      if (rechunkBuffer.isEmpty) ctx.finish()
-      else ctx.absorbTermination()
-    }
+        private var rechunkBuffer = ByteString.empty
 
-    private def rechunk(ctx: Context[ByteString]): SyncDirective = {
-      if (!ctx.isFinishing && ThreadLocalRandom.current().nextBoolean()) ctx.pull()
-      else {
-        val nextChunkSize =
-          if (rechunkBuffer.isEmpty) 0
-          else ThreadLocalRandom.current().nextInt(0, rechunkBuffer.size + 1)
-        val newChunk = rechunkBuffer.take(nextChunkSize)
-        rechunkBuffer = rechunkBuffer.drop(nextChunkSize)
-        if (ctx.isFinishing && rechunkBuffer.isEmpty) ctx.pushAndFinish(newChunk)
-        else ctx.push(newChunk)
+        private def rechunk() = {
+          if (!isClosed(in) && ThreadLocalRandom.current().nextBoolean()) pull(in)
+          else {
+            val nextChunkSize =
+              if (rechunkBuffer.isEmpty) 0
+              else ThreadLocalRandom.current().nextInt(0, rechunkBuffer.size + 1)
+            val newChunk = rechunkBuffer.take(nextChunkSize)
+            rechunkBuffer = rechunkBuffer.drop(nextChunkSize)
+            if (isClosed(in) && rechunkBuffer.isEmpty) completeStage()
+          }
+        }
+
+        override def onPush(): Unit = {
+          rechunkBuffer ++= grab(in)
+          rechunk()
+        }
+
+        override def onPull(): Unit = {
+          rechunk()
+        }
+
+        override def onUpstreamFinish(): Unit = {
+          if (rechunkBuffer.isEmpty) completeStage()
+          else completeStage()
+        }
       }
-    }
   }
 
-  val rechunk = Flow[ByteString].transform(() ⇒ new Rechunker).named("rechunker")
+  val rechunk = Flow[ByteString] via (new Rechunker).named("rechunker")
```

Comment:
In this case they are necessary because of `new`. I'dd just add the dot or switch to `new Rechunker().named(...)`